HereBeDragons-Pins-1.0.lua

```diff
@@ -514,7 +519,11 @@ function pins:AddWorldMapIconWorld(ref, icon, instanceID, x, y)
 
     if WorldMapButton:IsVisible() then
         local currentMapID, currentMapFloor = GetCurrentMapAreaID(), GetCurrentMapDungeonLevel()
-        if currentMapID and HBD.mapData[currentMapID] and HBD.mapData[currentMapID].instance == instanceID then
+        if currentMapID == -1 and GetCurrentMapContinent()==0 then
+            currentMapID = 0  -- force 0=Azeroth
+        end
+        local mapData = HBD.mapData[currentMapID]
+        if currentMapID and mapData and (mapData.instance == instanceID or mapData.floors[instanceID]) then
```

Comment:
This simply rejects Outland, Draenor and Maelstrom outright - without waiting for HBD:GetZoneCoordinatesFromWorld to return nils. It looks worse, but it's a tiny bit faster. I do agree, though, that the performance gain from skipping just one function call might not be huge.